File: 78-Subsets/78-Subsets.java
// Last updated: 7/2/2025, 11:54:23 PM
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        help(nums , 0, res , new ArrayList<Integer>());
        return res;
    }
    public void help(int[] nums ,int i,List<List<Integer>> res, ArrayList<Integer> lcl)
    {
        if(i == nums.length) 
        {
            res.add(new ArrayList<Integer>(lcl));
            return;
        }
        
        lcl.add(nums[i]);
        help(nums, i+1,res, lcl);
        lcl.remove(lcl.size()-1);
        help(nums ,  i+1,res,lcl);

    }
}

File: 78-Subsets/78-Subsets.cpp
// Last updated: 7/2/2025, 11:54:20 PM
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> lcl;

        help(nums, ans, lcl , 0);
        return ans;
    }

    void help(vector<int>& nums, vector<vector<int>> &ans, vector<int>& lcl, int idx){
        if(idx == nums.size()+1) return;
        ans.push_back(lcl);

        for(int i = idx ; i < nums.size(); i++)
        {
            lcl.push_back(nums[i]);
            help(nums, ans, lcl ,i+1);
            lcl.pop_back();

        }


    }
};